middleware/proxy/padding.go

```diff
@@ -0,0 +1,59 @@
+package proxy
+
+import (
+	"bytes"
+	"crypto/rand"
+	"encoding/base64"
+	"fmt"
+	"io"
+	"sync"
+)
+
+const errRandomnessGeneration = "failed to generate random bytes: %s"
+
+// Padding implements fixed-size, url-safe padding of strings
+type Padding struct {
+	Random  io.Reader
+	size    int
+	buffers sync.Pool
+}
+
+// NewPadding instantiates a new padding up to the given size (e.g. 256)
+func NewPadding(size uint16) *Padding {
+
+	// calculate the buffer we for a base64 representation that is "size" long
+	s := int(3.0 / 4.0 * float64(size))
+
+	return &Padding{
```

Comment:
Why all this? You just want to base64 encode some random string? I.e. `sync.Pool`?